ts/Core/Axis/GridAxis.ts

```diff
@@ -1246,38 +1246,48 @@ function onTickLabelFormat(ctx: AxisLabelFormatterContextObject): void {
  *       ticks and not the labels directly?
  */
 function onTrimTicks(this: Axis): void {
-    const axis = this;
-    const options = axis.options;
-    const gridOptions = options.grid || {};
-    const categoryAxis = axis.categories;
-    const tickPositions = axis.tickPositions;
-    const firstPos = tickPositions[0];
-    const lastPos = tickPositions[tickPositions.length - 1];
-    const linkedMin = axis.linkedParent && axis.linkedParent.min;
-    const linkedMax = axis.linkedParent && axis.linkedParent.max;
-    const min = linkedMin || axis.min;
-    const max = linkedMax || axis.max;
-    const tickInterval = axis.tickInterval;
-    const endMoreThanMin = (
-        firstPos < (min as any) &&
-        firstPos + tickInterval > (min as any)
-    );
-    const startLessThanMax = (
-        lastPos > (max as any) &&
-        lastPos - tickInterval < (max as any)
-    );
+    const axis = this,
+        options = axis.options,
+        gridOptions = options.grid || {},
+        categoryAxis = axis.categories,
+        tickPositions = axis.tickPositions,
+        firstPos = tickPositions[0],
+        secondPos = tickPositions[1],
+        lastPos = tickPositions[tickPositions.length - 1],
+        linkedMin = axis.linkedParent && axis.linkedParent.min,
+        linkedMax = axis.linkedParent && axis.linkedParent.max,
+        min = linkedMin || axis.min,
+        max = linkedMax || axis.max,
+        tickInterval = axis.tickInterval,
+        endMoreThanMin = (
+            min &&
```

Comment:
If `min` is 0, `endMoreThanMin` will end up as falsy. Not very likely that `min` is on January 1st 1970, but nonetheless. Use `isNumber(min)` instead.

Reply:
This also applies to the two more checks for `min` and `max` down below.